reactome_analysis_worker/reactome_analysis_worker/resources/r_code/riboseq_analyser.R

```diff
@@ -0,0 +1,671 @@
+    rna_samples <- exp_de %>%
+      filter(SeqType == "RNA") %>%
+      select(SampleID)
+
+    # Extract Sample IDs as a vector
+    rna_sample_ids <- rna_samples$SampleID
+
+    # Select columns from expression.data
+    expression.data.rna <- expression.data[, rna_sample_ids]
+
+    # Execute the analysis for RNA counts
+    ddsMat_rna <- DESeqDataSetFromMatrix(
+        countData = expression.data.rna,
+        colData=exp_de[which(exp_de$SeqType == "RNA"),],
+        design = design_R)
+
+    ddsMat_rna <- DESeq(ddsMat_rna)
+
+    # Extract results from the ddsMat object
+    res_rna <- results(ddsMat_rna, name="Group_d_vs_c")
+    res_rna <- lfcShrink(ddsMat_rna,coef="Group_d_vs_c", res=res_rna)
+
+
+    # FC analsysis for RIBO counts:
+    # Create filter using exp_de to find samples with SeqType  "RIBO"
+    ribo_data <- exp_de %>%
+      filter(SeqType == "RIBO") %>%
+      select(SampleID)
+
+    # Extract Sample IDs as a vector
+    ribo_sample_ids <- ribo_data$SampleID
+
+    # Select columns from expression.data
+    expression.data.ribo <- expression.data[, ribo_sample_ids]
+
+    # Getting FC for Ribo
+    ddsMat_ribo <- DESeqDataSetFromMatrix(
+        countData = expression.data.ribo,
+        colData = exp_de[which(exp_de$SeqType == "RIBO"),],
+        design = design_R)
+
+    ddsMat_ribo <- DESeq(ddsMat_ribo)
+
+    # Extract results from the ddsMat object
+    res_ribo <- results(ddsMat_ribo,name="Group_d_vs_c")
+    res_ribo <- lfcShrink(ddsMat_ribo,coef="Group_d_vs_c", res=res_ribo)
+
+    # Converts results for Translational Efficiency changes (TE) to dataframe
+    res$Identifier <- rownames(res)
+    res_combined <- as.data.frame(res)
+
+    # Extracts info from RNA results (res_rna)
+    res_rna_slim <- as.data.frame(res_rna) %>%
+      select(RNA_FC = log2FoldChange, RNA_padj = padj)
+
+    # Extracts info from RIBO results (res_ribo)
+    res_ribo_slim <- as.data.frame(res_ribo) %>%
+      select(RIBO_FC = log2FoldChange, RIBO_padj = padj)
+
+    res_rna_slim$Identifier <- rownames(res_rna_slim)
+    res_ribo_slim$Identifier <- rownames(res_ribo_slim)
+
+    # Combines the results into a single dataframe
+    res_combined <- res_combined %>%
+      left_join(res_ribo_slim, by = "Identifier") %>%
+      left_join(res_rna_slim, by = "Identifier")
+
+    # Assignes the Regulatory Mode to each gene in the results df
+    res_combined <- assign_Regmode(res_combined)
+
+    # Rename and reorder columns
+    names(res_combined)[names(res_combined) == "log2FoldChange"] <- "logFC"
+    names(res_combined)[names(res_combined) == "pvalue"] <- "P.Value"
+    names(res_combined)[names(res_combined) == "padj"] <- "adj.P.Val"
+    col_order <- c("Identifier", colnames(res_combined)[1:ncol(res_combined)])
+
+    return(res_combined[, col_order])
+}
+
+
+# Implementation of TerraPadog algorithm
+
+
+terapadog <- function (esetm = NULL, exp_de = NULL, paired = FALSE,
```

Comment:
Having the complete analysis code here is not ideal. This increases the complexity of our worker code quite a bit.

`terapadog` should be loaded from its own package. In case there is no package yet, there needs to be a discussion how this code will be maintained.

Reply:
I totally agree, with you. In case of packaging, maintaining could be done externally. (But will not be in our control) 
But generally, a Bioconductor package will be done anyways by @Gionmattia i guess.